hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2601,6 +2592,10 @@ public OzoneManagerHttpServer getHttpServer() {
         ServiceInfo.Builder peerOmServiceInfoBuilder = ServiceInfo.newBuilder()
             .setNodeType(HddsProtos.NodeType.OM)
             .setHostname(peerNode.getHostName())
+            // For now assume peer is at the same version.
+            // This field needs to be fetched from peer when rolling upgrades
+            // are implemented.
+            .setOmClientProtocolVersion(OZONE_OM_CLIENT_PROTOCOL_VERSION)
```

Comment:
Why we need this based on config value, why can't we do this based on version which we run.

As here if OM is not returning version means, newer S3G cannot talk to it, if any version is being returned it can talk to OM.

Reply:
And also with config it is confusing, Server is setting client protocol version.

Reply:
Server is setting the protocol version that it supports for clients. OM client protocol is the protocol between OM and the client as supported by the OM.
OM has a hard coded value that is decided at compile time based on the config set in code. This string is also made available on the client side to change based on the `OZONE_OM_CLIENT_PROTOCOL_VERSION_KEY`. For now this is the presence of the string but the version itself is setup to support semantic versioning.

Reply:
In general

Old client new server
New client old server are to be supported

But here we have a special case, where we are trying to protect new S3G talking to old OM. So, technically new RpcClient can talk to both old/new server. but here it is problem with new S3G talking to old OM. 

So, server setting protocol version that supports sounds odd to me.

Reply:
The server defines the protocol, it is saying that I am at `v2.0.0` which needs S3 auth to be set via proto messages. The client only speaks version `v2.0.0` and higher so if the OM is not reporting a version that it likes the client does not continue..

Reply:
The alternate client implementation which we decided against was for the client to discover the client version supported by OM and then change its behavior as per the spec.